llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -7638,7 +7644,7 @@ LegalizerHelper::lowerSADDO_SSUBO(MachineInstr &MI) {
   // (LHS) if and only if the other operand (RHS) is (non-zero) positive,
   // otherwise there will be overflow.
   auto ResultLowerThanLHS =
-      MIRBuilder.buildICmp(CmpInst::ICMP_SLT, BoolTy, Dst0, LHS);
```

Comment:
@arsenm the fact that we can't really change here is, `Dst0` is defined *twice*: one is in `MI`, and another one is either `MIRBuilder.buildAdd(Dst0, LHS, RHS);` or `MIRBuilder.buildSub(Dst0, LHS, RHS);`. This still causes the crash.